src/wallet/walletdb.cpp

```diff
@@ -1029,6 +1038,20 @@ std::unique_ptr<WalletDatabase> MakeDatabase(const fs::path& path, const Databas
         return nullptr;
     }
 
+    // A db already exists so format is set, but options also specifies the format, so make sure they agree
+    if (format && options.require_format && *format != options.require_format) {
```

Comment:
c4b6cf7ed72490df24a2e2d28d947002f3e6ef16

We could skip access to `format` value:
```suggestion
    if (format && options.require_format && format != options.require_format) {
```

Reply:
I think it is clearer this way.

Reply:
Is different approach to variables of the same type is clearer? Up to you :)

Reply:
Probably, it would be better to leave it as is for now, and apply changes in the follow up pull.

Reply:
Oh right, they're both `Optional`. Changed.